protocol/messenger_config.go

```diff
@@ -14,6 +14,10 @@ import (
 	"github.com/status-im/status-go/services/mailservers"
 )
 
+type MessengerSignalsHandler interface {
+	MessageDelivered(string, string)
+}
+
```

Comment:
I'm curious why you use an interface here, there doesn't seem to be any other structs that implement this except for a purpose created struct. Would something like the following also suit the situation?

```suggestion
type MessengerSignalsHandler func(string, string)
```

Then the consuming code could simply call something like:

```go
m.config.messengerSignalsHandler(message.LocalChatID, messageID)
```

Reply:
@Samyoul, thanks for reviewing!
I used an interface because I thought it will be more consistent with existing `services/ext/signals.go` content. All functions there grouped logically with the help of empty structs. Since `delivered` is a first messenger's signal and probably not the last one,  same logic were used. I created empty struct and function of it.
And in `services/ext/service.go` I took `EnvelopeSignalHandler{}` as an example and used `MessengerSignalsHandler{}` similarly.

Overall this empty structs approach seemed unfamiliar to me, probably due to lack of golang experience but I think it will work well for Messenger when more signals added. And most probably they will be added when more logic moved out of `status-react`. 

Wdyt?

Reply:
Thank you for your reasoning. I can see your approach.

Personally I think that an interface with a single struct implementation wrapping a single function is unnecessary complexity. `ext.EnvelopeEventsHandler{}` struct is similar that it has no state fields, but the `ext.EnvelopeEventsHandler{}` interface does have multiple methods and multiple implementations (it also confusingly has 4 declarations), and the `protocol.EnvelopeEventsInterceptor{}` implementation has state fields.

> And most probably they will be added when more logic moved out of status-react.

This is the best reason for keeping your current approach, if we plan on implementing multiple message signal handler functions in the near future then it makes sense to use a struct to load in all the functions at the same time.

Reply:
>Personally I think that an interface with a single struct implementation wrapping a single function is unnecessary complexity
Totally agree)

>Would the MessengerSignalsHandler {} interface have more than one implementation?
Probably not. Maybe some can be added for testing if needed.

@cammellos, do you know if Messenger will need more signals to client when logic moved out of `status-react`? 
If there are no events we can think about right now, I'd be happy to remove unnecessary interface and use the approach suggested by @Samyoul

Reply:
mmh, tough question :)
I can't think of anything at the moment, signals tends to be pretty stable, though maybe when we move some mailserver logic we might have to create new signals, hard to say.
I don't mind either approaches here myself, so it's completely up to you two, as you prefer (I tend to favor Sam's point, but there's little complexity added and we might add more signals, so I think both are acceptable)

Reply:
Thanks! If we aren't sure - I'd also prefer to simplify the code :)